intake/source/csv.py

```diff
@@ -123,7 +123,10 @@ def files(self):
         if self.pattern is None and not glob_in_path:
             self._files = urlpath
         else:
-            self._files = get_fs_token_paths(urlpath)[2]
+            protocol = split_protocol(urlpath[0])[0]
+            self._files = get_fs_token_paths(urlpath, storage_options=self._storage_options)[2]
+            if protocol:
+                self._files = [f"{protocol}://{fn}" for fn in self._files]
```

Comment:
Unfortunately, not all implementations strip the protocol the same way, so you'll need `fs.unstrip_protocol`.

Other options would be:
- add `protocol=` to the storage_options, if it's not already there
- to store the fs we infer
- to use fs.open_files and store those OpenFiles
I don't immediately know how much further knock-on changes either of those would cause.

Reply:
I think I prefer bullet 1.

Reply:
Bullet 1 definitely looks best to me too. Unfortunately this wasn't working for me:

```python
In [4]: from fsspec.core import get_fs_token_paths

In [5]: url = "s3://test-for-intake/*.csv"

In [6]: so = {"s3": {"anon": True}}

In [7]: get_fs_token_paths(url, storage_options=so)
Out[7]:
(<s3fs.core.S3FileSystem at 0x151378f40>,
 '081ddbc04ab2fb22c401dcebc75ec0bc',
 ['test-for-intake/intake_test.csv'])

In [8]: so["protocol"] = "s3"

In [9]: so
Out[9]: {'s3': {'anon': True}, 'protocol': 's3'}

In [10]: get_fs_token_paths(url, storage_options=so)
Out[10]:
(<s3fs.core.S3FileSystem at 0x151378f40>,
 '081ddbc04ab2fb22c401dcebc75ec0bc',
 ['test-for-intake/intake_test.csv'])

In [11]: get_fs_token_paths(url, storage_options=so, protocol="s3")
Out[11]:
(<s3fs.core.S3FileSystem at 0x151378f40>,
 '081ddbc04ab2fb22c401dcebc75ec0bc',
 ['test-for-intake/intake_test.csv'])
```

Reply:
Ah right. So those paths are correct, but pandas won't even ask fsspec to handle them unless there's a protocol:// in place :|
Back to unstrip, then.